ProcessLib/HT/HTProcess.cpp

```diff
@@ -24,6 +24,42 @@ namespace ProcessLib
 {
 namespace HT
 {
+void checkMPLProperties(MeshLib::Mesh const& mesh,
+                        HTProcessData const& process_data)
+{
+    DBUG("Check the media properties of HT process ...");
+
+    std::array const requiredPropertyMedium = {
+        MaterialPropertyLib::PropertyType::porosity,
+        MaterialPropertyLib::PropertyType::permeability};
+
+    std::array const requiredPropertyLiquidPhase = {
+        MaterialPropertyLib::PropertyType::viscosity,
+        MaterialPropertyLib::PropertyType::density,
+        MaterialPropertyLib::PropertyType::specific_heat_capacity};
+
+    std::array const requiredPropertySolidPhase = {
+        MaterialPropertyLib::PropertyType::specific_heat_capacity,
+        MaterialPropertyLib::PropertyType::density,
+        MaterialPropertyLib::PropertyType::storage};
+
+    for (auto const& element : mesh.getElements())
```

Comment:
Not sure here. If we write `for (auto e : mesh.getElements())`, `e` has type `Element*` (optionally const qualified). In the current case there is additional reference... Not sure what a compiler will do out of `Element*&`...